client/allocrunner/taskrunner/task_runner_hooks.go

```diff
@@ -109,6 +109,8 @@ func (tr *TaskRunner) initHooks() {
 
 	// If there are templates is enabled, add the hook
 	if len(task.Templates) != 0 {
+		tr.runnerHooks = append(tr.runnerHooks, newConsulHook(hookLogger, tr))
```

Comment:
If someone has configured a Consul identity _without_ a `template` block (so they can make their own API calls to Consul, for example), they wouldn't get the environment variable / file because this is guarded behind this conditional. I think we probably want to move this outside the `len(task.Templates) != 0` check, but still before we call the template.